src/components/mainComponent/MainComponent.spec.js

```diff
@@ -19,7 +20,7 @@ describe('Main Component', () => {
   it('renders a new page for human vs. computer after start button is click', () => {
     const wrapper = mount(<Main />);
 
-    wrapper.find('.computer-page').simulate('click');
+    wrapper.find(ButtonComponent).at(1).simulate('click');
```

Comment:
Also from our conversation from yesterday, remember that you don't want to select React components directly. Instead, select them by their class (e.g. `wrapper.find('.computer-page').simulate('click');`)

Reply:
What is the issue with finding the component by its type? At the unit level, coupling to the implementation is common. i.e. This doesn't feel fundamentally different than leverage a fake or other form of test double.

Reply:
It's not so much the `find(ButtonComponent)` that is a problem, it is the `at(1)` that bothers me - @ericmeyer is right, testing `ButtonComponent` directly at the unit level is totally fine, and I do that pretty often. The problem I see here is you are mounting `<Main />` and then testing the `ButtonComponent` through main. So, for example, let's say you add another button at the top of the page to navigate to "About Us and Why We Think TTT is So Cool". You have just broken the test for starting a computer vs. human game randomly since the index of that button component has now changed to "2".

Reply:
That is why I like to select on classes or some data-id - you can totally rearrange the layout of your page, and your tests of business logic still pass